packages/yourloops/lib/auth/hook.tsx

```diff
@@ -518,19 +246,68 @@ export function AuthContextImpl(api: AuthAPI): AuthContext {
 
   const redirectToProfessionalAccountLogin = (): void => window.location.assign(`${appConfig.API_HOST}/auth/oauth/login`);
 
-  React.useEffect(initHook, [historyHook, pathname, traceToken, isAuthInProgress, setAuthInfos]);
+  /*******************************************************/
+  /****** AUTH 0 HACK FOR LOGIN/LOGOUT AND SIGNUP *******/
+  /*****************************************************/
+
+  const mapAuth0UserToIUser = useMemo<IUser>(() => {
+    let user = {};
+    if (auth0user && auth0user.sub) {
+      const parsedSub = auth0user.sub.split("|");
+      const id = parsedSub[1];
+      user = {
+        role: auth0user["http://your-loops.com/roles"][0],
+        userid: id,
+        emailVerified: auth0user.email_verified,
+        username: auth0user.email || "",
+      };
+    }
+    return user as IUser;
+  }, [auth0user]);
+
+  const getUserInfo = useCallback(async () => {
+    try {
+      if (auth0user) {
+        const user = new User(mapAuth0UserToIUser);
+        const sessionToken = await api.getShorelineAccessToken(auth0user.email as string);
+        const traceToken = uuidv4();
+        const updatedUser = await api.getUserInfo({ user, sessionToken, traceToken });
+        setUser(updatedUser);
+        setSessionToken(sessionToken);
+        setTraceToken(traceToken);
+      }
+    } catch (err) {
+      log.error(err);
+    }
+  }, [api, auth0user, mapAuth0UserToIUser]);
+
+  const logout = async (): Promise<void> => {
+    try {
+      if (window.cleanBlipReduxStore) {
+        window.cleanBlipReduxStore();
+      }
+      zendeskLogout();
+      await auth0logout({ returnTo: window.location.origin });
+    } catch (err) {
+      log.error("logout", err);
+    }
+  };
+
+  useEffect(() => {
+    const getAccessToken = async () => getAccessTokenSilently();
+    (async () => {
+      if (isAuthenticated && !user) {
+        HttpService.setAccessToken(getAccessToken);
```

Comment:
`setAccessToken` feels weird and should be something like `setGetAccessTokenMethod`. Do you see what I mean?